app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/SmoothnessItem.kt

```diff
@@ -0,0 +1,125 @@
+package de.westnordost.streetcomplete.quests.smoothness
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.smoothness.Smoothness.*
+import de.westnordost.streetcomplete.view.image_select.Item
+
+fun Array<Smoothness>.toItems(surface: String, highway: String) = this.mapNotNull { it.asItem(surface, highway) }
+
+// return null if not a valid combination
+fun Smoothness.asItem(surface: String, highway: String): Item<Smoothness>? {
+    val imageResId = getImageResId(surface) ?: return null
+    val descriptionResId = getDescriptionResId(surface, highway) ?: return null
+    return Item(this, imageResId, getTitleResId(), descriptionResId)
+}
+
+fun Smoothness.getTitleResId() = when (this) {
+    EXCELLENT -> R.string.quest_smoothness_title_excellent
+    GOOD -> R.string.quest_smoothness_title_good
+    INTERMEDIATE -> R.string.quest_smoothness_title_intermediate
+    BAD -> R.string.quest_smoothness_title_bad
+    VERY_BAD -> R.string.quest_smoothness_title_very_bad
+    HORRIBLE -> R.string.quest_smoothness_title_horrible
+    VERY_HORRIBLE -> R.string.quest_smoothness_title_very_horrible
+    IMPASSABLE -> R.string.quest_smoothness_title_impassable
+}
+
+fun Smoothness.getDescriptionResId(surface: String, highway: String) = when (this) {
+    EXCELLENT -> when (surface) {
+        // no "excellent" for roads with paving stones
```

Comment:
it is possible to have excellent surface also on `surface=paving_stones`

Reply:
I think this was already discussed, I didn't want to have it for roads because I've only seen it for foot paths (or anything where there are no heavy vehicles potentially damaging the stones).
But it can be allowed again of course

Reply:
Apologies for potentially ignorant questions: why is the comment still there if paving stones have been returned. Also, am I missing concrete lots of places?